gapic/src/main/com/google/gapid/perfetto/Unit.java

```diff
@@ -156,13 +173,30 @@ public String format(double value) {
     }
   }
 
+  public boolean convertible() {
+    return formatter.convertible();
+  }
+
+  public int findOptimalOffset(double value) {
+    return formatter.findOptimalOffset(value);
+  }
+
+  public String convertTo(double value, int offset) {
+    return formatter.convertTo(value, offset);
+  }
+
```

Comment:
I think a better interface for this would be:

```
public Unit withFixedScale(double representativeValue) {
  return new Unit(name, formatter.withFixedScale(representativeValue));
}

private static interface Formatter {
  // ...
  public default Formatter withFixedScale(double representativeValue) { return this; }
}
```

For formatters with a scale, it can compute the offset and then return an anonymous `Formatter` implementation with the simplified formatting. You can then hide the internal implementation details (i.e. the "offset" discussion) inside the `Formatter` implementation.

Reply:
Thanks for the suggestion! Just updated with this change :)